recipes/qcustomplot/all/conanfile.py

```diff
@@ -39,19 +41,19 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
-        # FIXME: we shouldn't have to force shared in qt, but config file
-        #        generated by conan in qt static is likely broken, or maybe env vars.
-        self.options["qt"].shared = True
 
     def requirements(self):
-        if Version(self.version) >= "2.0.0":
-            self.requires("qt/6.4.1")
-        else:
-            self.requires("qt/5.15.7")
+        self.requires("qt/6.4.2", transitive_headers=True, transitive_libs=True)
+
         if self.options.with_opengl and self.settings.os == "Windows":
             self.requires("opengl/system")
 
+    def build_requirements(self):
+        self.tool_requires("qt/6.4.2")
```

Comment:
```suggestion
        self.tool_requires("qt/<host_version>")
```
`<host_version>` is an actual thing in Conan. :) It's recommended to use it so if someone downstream overrides the Qt version, it won't cause a conflict.
Also needs a `required_conan_version = ">=1.56.0 <2 || >=2.0.6"`.